### ISSUE TITLE
Support edgecore run in container

### ISSUE THREAD

**Issue by daixiang0** [kubeedge/kubeedge#2177]
**What would you like to be added/modified**:

Support edgecore run in container.

**Why is this needed**:

Now we have `hack/local-up-cluster.sh` to run a cluster including one edge node in a single machine, it would be useful for multi-edges cluster test for both develop and benchmark in a single machine.

To support it, we need optimize process check way since now we only check process exist whatever it exists in container or host.

**Comment by kevin-wangzefeng** [comment]
We already supported edgecore running inside container for tests IMO, please add the details you are proposing to do here.
xref: 
- https://github.com/kubeedge/kubeedge/blob/master/docs/proposals/perf.md
- https://github.com/kubeedge/kubeedge/blob/master/build/edge/Dockerfile

**Comment by daixiang0** [comment]
Yep, i would update desribtion since this container can succeed in build but can not run any pods.

**Comment by fisherxu** [comment]
`Local-up` cluster used to fast deploy and debug for developers in local host, and for general use case, one edgenode is enough. If it's necessary to run edgecore as container, we can add the flag to control.

**Comment by daixiang0** [comment]
For edge side network test, one is not enough. 

Also, the root problem is that edgecore can run in container but failed to start any pods.

**Comment by luogangyi** [comment]
I have met the same problem and did some digging.
It seems kubelet already support running itself in container, see 
https://github.com/kubernetes/kubernetes/issues/4869

I think it requires some updates and configuration.

**Comment by yusiwen** [comment]
Any updates?
